stdlib/assembly.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Why apache?

Reply:
Could be MIT too. The goal is to avoid license baggage for users. They may want to keep their deployed codebase with less restrictions than GPL-3.0.

Reply:
One thing I really don't like about Apache license is this bit:

> You must cause any modified files to carry prominent notices stating that You changed the files

I.e. it **requires** the source to be annotated with extensive info about who changed it. I think that such metadata belongs in the version control, not in the file. It would be pretty tedious to have to keep it in the source.

Reply:
I think it just means stating it was changed and not what has been. Though in practice this does not take place in the original source repo, just in forks.

Also note Rust uses a dual MIT/Apache2 license and a large part of the Ethereum ecosystem uses Apache2.